src/spec/template_spec.js

```diff
@@ -0,0 +1,333 @@
+    setupTemplateMenuSuccessResponse();
+    var body = "Test content body";
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.fillTemplateBodySection(body);
+    templatePage.clickSaveTemplateButton();
+    
+    // Assert
+    expect(browser.getCurrentUrl()).toEqual(browser.baseUrl + '/#/templates');
+  });
+
+  it('should save template in edit mode and fail', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/32e93fa4-c8f4-467a-a2bb-da95f72a27b4');
+    setupTemplateResponse();
+    setupEditTemplateErrorResponse();
+    setupTemplateMenuSuccessResponse();
+    var body = "Test content body";
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.fillTemplateBodySection(body);
+    templatePage.clickSaveTemplateButton();
+    
+    // Assert
+    expect(templatePage.isErrorModalDisplayed()).toBeTruthy();
+  });
+
+  it('should create a new template and fail', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/new');
+    setupTemplateResponse();
+    setupCreateNewTemplateFailResponse();
+    setupTemplateMenuSuccessResponse();
+    var templateName = "Test template";
+    var templateFromEmail = "test@test.test";
+    var templateFromName = "test name";
+    var templateSubject = "test subject";
+    var templateBody = "Template body test"
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.fillTemplateNameInput(templateName);
+    templatePage.fillTemplateFromEmailInput(templateFromEmail);
+    templatePage.fillTemplateFromNameinput(templateFromName);
+    templatePage.fillTemplateSubjectInput(templateSubject);
+    templatePage.fillTemplateBodySection(templateBody);
+    templatePage.clickSaveTemplateButton();
+    
+    // Assert
+    expect(templatePage.isErrorModalDisplayed()).toBeTruthy();
+  });
+
+  it('should create a new template and redirect to template list', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/new');
+    setupTemplateResponse();
+    setupCreateTemplateSuccessResponse();
+    setupTemplateMenuSuccessResponse();
+    var templateName = "Test template";
+    var templateFromEmail = "test@test.test";
+    var templateFromName = "test name";
+    var templateSubject = "test subject";
+    var templateBody = "Template body test"
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.fillTemplateNameInput(templateName);
+    templatePage.fillTemplateFromEmailInput(templateFromEmail);
+    templatePage.fillTemplateFromNameinput(templateFromName);
+    templatePage.fillTemplateSubjectInput(templateSubject);
+    templatePage.fillTemplateBodySection(templateBody);
+    templatePage.clickSaveTemplateButton();
+    
+    // Assert
+    expect(browser.getCurrentUrl()).toEqual(browser.baseUrl + '/#/templates');
+  });
+
+  it('should not allow to create a new template if the form is not completed', () => {
```

Comment:
It is more common the usage of _when_ in place of _if_

Reply:
👍  thx